src/Models/AZ3166Device.ts

```diff
@@ -315,118 +320,194 @@ export class AZ3166Device implements Device {
+            });
 
-      if (!selection) {
-        return false;
-      }
+        if (!selection) {
+          return false;
+        }
 
-      if (selection.detail === 'Input another...') {
-        const option: vscode.InputBoxOptions = {
-          value:
-              'HostName=<Host Name>;DeviceId=<Device Name>;SharedAccessKey=<Device Key>',
-          prompt: `Please input device connection string here.`,
-          ignoreFocusOut: true
-        };
+        if (selection.detail === 'Input another...') {
+          const option: vscode.InputBoxOptions = {
+            value:
+                'HostName=<Host Name>;DeviceId=<Device Name>;SharedAccessKey=<Device Key>',
+            prompt: `Please input device connection string here.`,
+            ignoreFocusOut: true
+          };
+
+          deviceConnectionString = await vscode.window.showInputBox(option);
+          if (!deviceConnectionString) {
+            return false;
+          }
+          if ((deviceConnectionString.indexOf('HostName') === -1) ||
+              (deviceConnectionString.indexOf('DeviceId') === -1) ||
+              (deviceConnectionString.indexOf('SharedAccessKey') === -1)) {
+            throw new Error(
+                'The format of the IoT Hub Device connection string is invalid. Please provide a valid Device connection string.');
+          }
+        }
 
-        deviceConnectionString = await vscode.window.showInputBox(option);
         if (!deviceConnectionString) {
           return false;
         }
-        if ((deviceConnectionString.indexOf('HostName') === -1) ||
-            (deviceConnectionString.indexOf('DeviceId') === -1) ||
-            (deviceConnectionString.indexOf('SharedAccessKey') === -1)) {
-          throw new Error(
-              'The format of the IoT Hub Device connection string is invalid. Please provide a valid Device connection string.');
+
+        console.log(deviceConnectionString);
+
+        // Set selected connection string to device
+        let res: boolean;
+        const plat = os.platform();
+        if (plat === 'win32') {
+          res = await this.flushDeviceConfig(
+              deviceConnectionString, options.ConnectionString);
+        } else {
+          res = await this.flushDeviceConfigUnix(
+              deviceConnectionString, options.ConnectionString);
+        }
+
+        if (res === false) {
+          return false;
+        } else {
+          vscode.window.showInformationMessage(
+              'Configure Device connection string successfully.');
+          return true;
         }
+      } catch (error) {
+        throw error;
       }
+    } else {
+      try {
+        const option: vscode.InputBoxOptions = {
+          value:
+              '19e25a259d0c2be03a02d416c05c48ccd0cc7d1743458aae1cb488b074993eae',
+          prompt: `Please input UDS here.`,
+          ignoreFocusOut: true,
+          validateInput: (UDS: string) => {
+            if (/^([0-9a-f]){64,64}$/i.test(UDS) === false) {

```

Comment:
`/^([0-9a-f]){64,64}$/i` is the same as `/^([0-9a-f]){64}$/i`, `/^([0-9a-f]){64}$/i` makes more sense.